src/main/java/com/firstspringmvcapp/controllers/AuthController.java

```diff
@@ -0,0 +1,121 @@
+import com.firstspringmvcapp.services.PeopleService;
+import com.firstspringmvcapp.util.PersonErrorResponse;
+import com.firstspringmvcapp.util.PersonNotCreatedException;
+import com.firstspringmvcapp.util.PersonValidator;
+
+import java.util.List;
+import java.util.Map;
+
+@Controller
+@RequestMapping("/auth")
+public class AuthController {
+
+    private final PersonValidator personValidator;
+    private final PeopleService peopleService;
+    private final AdminService adminService;
+    private final ModelMapper modelMapper;
+    private final JWTUtil jwtUtil;
+    private final AuthenticationManager authManager;
+
+    @Autowired
+    public AuthController(PersonValidator personValidator, PeopleService peopleService, AdminService adminService,
+                          ModelMapper modelMapper, JWTUtil jwtUtil, AuthenticationManager authManager) {
+        this.personValidator = personValidator;
+        this.peopleService = peopleService;
+        this.adminService = adminService;
+        this.modelMapper = modelMapper;
+        this.jwtUtil = jwtUtil;
+        this.authManager = authManager;
+    }
+
+    @GetMapping("/login")
+    public String loginPage() {
+        return "auth/login";
+    }
+
+    @GetMapping("/registration")
+    public String registrationPage(@ModelAttribute("personDTO") PersonDTO personDTO) {
+        return "auth/registration";
+    }
+
+    @PostMapping("/registration")
+    public String performRegistration(@Valid PersonDTO personDTO,
+                                                   BindingResult bRes) {
+        Person person = convertToPerson(personDTO);
+        personValidator.validate(person, bRes);
+
+        if (bRes.hasErrors()) {
+            StringBuilder errorMessage = new StringBuilder();
+
+            List<FieldError> errors = bRes.getFieldErrors();
+            for (FieldError error : errors) {
+                errorMessage.append(error.getField())
+                        .append(" - ").append(error.getDefaultMessage())
+                        .append(";");
+            }
+            throw new PersonNotCreatedException(errorMessage.toString());
+        }
+        peopleService.save(person);
+
+        String token = jwtUtil.generateToken(person.getEmail());
+//        return Map.of("jwt-token", token);
+        return "redirect:/people";
+    }
+
+    @ExceptionHandler
+    private ResponseEntity<PersonErrorResponse> handleException(PersonNotCreatedException e) {
+        PersonErrorResponse response = new PersonErrorResponse(
+                e.getMessage(), System.currentTimeMillis()
+        );
+
+        return new ResponseEntity<>(response, HttpStatus.BAD_REQUEST);
+    }
+
+    @GetMapping("/admin")
+    public String adminPage() {
+        adminService.doAdminAction();
+        return "auth/admin";
+    }
+
+    @PostMapping("/login")
+    public Map<String, String> performLogin(@RequestBody AuthenticationDTO authenticationDTO) {
```

Comment:
What do you think about adding spring security configuration and enable login 
https://www.baeldung.com/spring-security-login#3-configuration-for-form-login